cpp/arcticdb/util/name_validation.cpp

```diff
@@ -118,8 +119,14 @@ void verify_library_path_part(const std::string& library_part, char delim) {
     }
 }
 
-void verify_library_path_on_write(const StringId& library_path) {
+void verify_library_path_on_write(const std::shared_ptr<Store>& store, const StringId& library_path) {
```

Comment:
Avoid passing const ref to std::shared_pointer in favor of regular const pointer or const ref, unless the code is interested in some particular property of the shared_pointer (for example it want's to copy the shared pointer).

see [here](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-smart) and [here](https://herbsutter.com/2013/06/05/gotw-91-solution-smart-pointer-parameters/)